Tools/build/generate_sbom.py

```diff
@@ -106,13 +108,106 @@ def filter_gitignored_paths(paths: list[str]) -> list[str]:
     return sorted([line.split()[-1] for line in git_check_ignore_lines if line.startswith("::")])
 
 
+def discover_pip_sbom_package(sbom_data: dict[str, typing.Any]) -> None:
+    """pip is a part of a packaging ecosystem (Python, surprise!) so it's actually
+    automatable to discover the metadata we need like the version and checksums
+    so let's do that on behalf of our friends at the PyPA.
+    """
+    global PACKAGE_TO_FILES, CPYTHON_ROOT_DIR
+
+    ensurepip_bundled_dir = CPYTHON_ROOT_DIR / "Lib/ensurepip/_bundled"
+    pip_wheels = []
+
+    # Find the hopefully one pip wheel in the bundled directory.
+    for wheel_filename in os.listdir(ensurepip_bundled_dir):
+        if wheel_filename.startswith("pip-"):
+            pip_wheels.append(wheel_filename)
+    if len(pip_wheels) != 1:
+        print("Zero or multiple pip wheels detected in 'Lib/ensurepip/_bundled'")
+        exit(1)
```

Comment:
> [exit](http://docs.python.org/library/constants.html#exit) is a helper for the interactive shell - [sys.exit](http://docs.python.org/library/sys.html#sys.exit) is intended for use in programs.

https://stackoverflow.com/a/6501134/724176

```suggestion
        sys.exit(1)
```

Reply:
TIL about the difference between the two, thank you!

Reply:
There's also `raise SystemExit()`, which is basically the same as `sys.exit()` and you can skip the `import` but 🤷 

https://adamj.eu/tech/2021/10/10/the-many-ways-to-exit-in-python/